### PR TITLE
chore: Move estimator contracts to test contracts

### LINKED ISSUE #7070: Building estimator test contract
The build script at `runtime/runtime-params-estimator/test-contract/build.sh` currently builds the test contract used for estimation. This script is invoked before actually running estimations but the workflow is a bit iffy. Let's clean this up.

See also this comment for context and ideas how to clean it up: https://github.com/near/nearcore/issues/6822#issuecomment-1160338075

Ans also related to this, after #7067, the script became so simple that having a script for it just makes no sense.

### PR DESCRIPTION
**Pull request by jakmeier**
The test contracts used by the estimator used to live in
nearcore/runtime/runtime-params-estimator and was built with custom
bash scripts. Now they are along side other test contracts and reuse the
build infrastructure there.

Resolves #7070

### CODE REVIEW

**Comment by jakmeier** [comment]
Looking to remove the `build.sh` as discussed, I didn't really see a reason why the estimator contract needs to be built different than other test contracts. Unless I missed something, I believe this is the cleanest approach here.

I also thought about providing the contract as an input to the estimator. I think it might make sense (in a future PR) to add a CLI argument to overwrite the estimator contract with some WASM. But in the normal case, I think I just want the estimator to build the default contract to make sure it is there in the newest version.

**Comment by matklad** [comment]
> I didn't really see a reason why the estimator contract needs to be built different than other test contracts. Unless I missed something, I believe this is the cleanest approach here.

One thing that's different between estimator contracts and other cotnracts is that estimator contracts shouldn't in general change. EG, it would be wrong to use our standard contract for estimation, as it changes over time. 

But I think it should be fine to re-use build infrastructure here, yeah.

One thing I am wondering if we should group estimator-related contracts into a sub-directory with an extra note of "don't gratitiously refactor this please"?

File: runtime/near-test-contracts/src/lib.rs [hunk 1]
```diff
@@ -79,6 +79,23 @@ pub fn fuzzing_contract() -> &'static [u8] {
     CONTRACT.get_or_init(|| read_contract("contract_for_fuzzing_rs.wasm")).as_slice()
 }
 
+/// Smallest (reasonable) contract possible to build from Rust code.
+pub fn smallest_rs_contract() -> &'static [u8] {
+    static CONTRACT: OnceCell<Vec<u8>> = OnceCell::new();
+    CONTRACT.get_or_init(|| read_contract("smallest_contract")).as_slice()
```
Comment (matklad) [inline runtime/near-test-contracts/src/lib.rs:85]: Perhaps just use `wat_contract` here? using Rust for smallest contract feels counter-productive. 

Also, could you update comment on `wat_contract` as a drive-by? It's copy-paste error

**Review by matklad — APPROVED** [review]
LGTM

Do we need this wasm in repo?

runtime/near-test-contracts/smallest-contract-rs/res/smallest_contract.wasm

Seems like we can drop it?

File: runtime/near-test-contracts/src/lib.rs (hunk 1 above)
Comment (jakmeier) [reply runtime/near-test-contracts/src/lib.rs]: Yes good call, now I just removed the smallest contract crate entirely.

**Comment by jakmeier** [comment]
> One thing I am wondering if we should group estimator-related contracts into a sub-directory with an extra note of "don't gratitiously refactor this please"?

After removing the smallest contract, there is only one contract that would be in such a subdirectory. Which is already called estimator contract. And I think future contracts for estimation will probably be constructed from WAT and therefore don't need a Rust crate in near-test-contracts. So I think we don't need the subdirectory right now.
